modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,758 @@
+"{ mtcnnom    | mtcnn-o.xml               | Path to OpenVINO MTCNN O (Output) detection model (.xml)}"
+"{ mtcnnod    | CPU                       | Target device for the MTCNN O (e.g. CPU, GPU, VPU, ...) }"
+"{ thrp       | 0.6                       | MTCNN P confidence threshold}"
+"{ thrr       | 0.7                       | MTCNN R confidence threshold}"
+"{ thro       | 0.7                       | MTCNN O confidence threshold}"
+"{ half_scale | false                     | MTCNN P use half scale pyramid}"
+;
+
+namespace {
+std::string weights_path(const std::string& model_path) {
+    const auto EXT_LEN = 4u;
+    const auto sz = model_path.size();
+    CV_Assert(sz > EXT_LEN);
+
+    const auto ext = model_path.substr(sz - EXT_LEN);
+    CV_Assert(cv::toLowerCase(ext) == ".xml");
+    return model_path.substr(0u, sz - EXT_LEN) + ".bin";
+}
+//////////////////////////////////////////////////////////////////////
+} // anonymous namespace
+
+namespace custom {
+namespace {
+
+// Define custom structures and operations
+#define NUM_REGRESSIONS 4
+#define NUM_PTS 5
+
+struct BBox {
+    double x1;
+    double y1;
+    double x2;
+    double y2;
+
+    cv::Rect getRect() const { return cv::Rect(static_cast<int>(x1), static_cast<int>(y1), static_cast<int>(x2 - x1), static_cast<int>(y2 - y1)); }
+
+    BBox getSquare() const {
+        BBox bbox;
+        double bboxWidth = x2 - x1;
+        double bboxHeight = y2 - y1;
+        double side = std::max(bboxWidth, bboxHeight);
+        bbox.x1 = static_cast<double>(x1) + (bboxWidth - side) * 0.5;
+        bbox.y1 = static_cast<double>(y1) + (bboxHeight - side) * 0.5;
+        bbox.x2 = bbox.x1 + side;
+        bbox.y2 = bbox.y1 + side;
+        return bbox;
+    }
+};
+
+struct Face {
+    BBox bbox;
+    double score;
+    double regression[NUM_REGRESSIONS];
+    double ptsCoords[2 * NUM_PTS];
+
+    static void applyRegression(std::vector<Face>& faces, bool addOne = false) {
+        for (size_t i = 0; i < faces.size(); ++i) {
+            double bboxWidth =
+                faces[i].bbox.x2 - faces[i].bbox.x1 + static_cast<double>(addOne);
+            double bboxHeight =
+                faces[i].bbox.y2 - faces[i].bbox.y1 + static_cast<double>(addOne);
+            faces[i].bbox.x1 = faces[i].bbox.x1 + static_cast<double>(faces[i].regression[1]) * bboxWidth;
+            faces[i].bbox.y1 = faces[i].bbox.y1 + static_cast<double>(faces[i].regression[0]) * bboxHeight;
+            faces[i].bbox.x2 = faces[i].bbox.x2 + static_cast<double>(faces[i].regression[3]) * bboxWidth;
+            faces[i].bbox.y2 = faces[i].bbox.y2 + static_cast<double>(faces[i].regression[2]) * bboxHeight;
+        }
+    }
+
+    static void bboxes2Squares(std::vector<Face>& faces) {
+        for (size_t i = 0; i < faces.size(); ++i) {
+            faces[i].bbox = faces[i].bbox.getSquare();
+        }
+    }
+
+    static std::vector<Face> runNMS(std::vector<Face>& faces, double threshold,
```

Comment:
const for parameters. Move const parameters to first places in signature.

Reply:
what about useMin default false?

Reply:
This comment was about `threshold`